lib/shoulda/matchers/active_record/association_matchers/join_table_matcher.rb

```diff
@@ -37,8 +37,16 @@ def join_table_option_correct?
             end
           end
 
+          def tables_and_views(table_or_view)
```

Comment:
What do you think about having this method return `connection.data_sources` or `connection.tables` instead of a boolean? Also, let's put this in RailsShim -- we have a collection of methods there that deal with compatibility between Rails versions.